projects/plugins/boost/app/lib/critical-css/source-providers/providers/Archive_Provider.php

```diff
@@ -92,13 +92,32 @@ public static function get_post_types() {
 			array(
 				'public'      => true,
 				'has_archive' => true,
-			)
+			),
+			'objects'
 		);
 		unset( $post_types['attachment'] );
 
 		$post_types = array_filter( $post_types, 'is_post_type_viewable' );
 
-		return apply_filters( 'jetpack_boost_critical_css_post_types', $post_types );
+		$provider_post_types = array();
+		// Generate a name => name array for backwards compatibility.
+		foreach ( $post_types as $post_type ) {
+			$provider_post_types[ $post_type->name ] = $post_type->name;
+		}
+
+		return apply_filters(
+			'jetpack_boost_critical_css_post_types_archives',
+			apply_filters_deprecated(
+				'jetpack_boost_critical_css_post_types',
+				array(
+					$provider_post_types,
+					$post_types,
+				),
+				'3.4.0',
+				'jetpack_boost_critical_css_post_types_archives'
+			),
+			$post_types
+		);
```

Comment:
> How `apply_filters_deprecated` is documented, you do not need to make an `apply_filters` call. Only one call to `apply_filters_deprecated` would also apply the new filter.

If we call the deprecated filter, there will be a deprecation warning. I wanted to avoid that so I added the `apply_filters` before it and used the new filter to ignore post types.

Now if someone uses the old filter, they would get a warning and everything would still work.

> The arguments aren't the same in the old and new filter. Do we need to change the argument types?

The only difference here is the introduction of a new argument. The original one has a new name, but the type is the same - key => value pair. But seeing as we're deprecating it, we don't need `$post_types` in the old one. Thanks for making me think about this 😅

Reply:
To clarify - previously, there was a bug with [`$post_types`](https://github.com/Automattic/jetpack/pull/37095/files#diff-86d81f3788f3e6c150a5c0b0671805258476cba0998eaaf019a32f4f615edfcdR91) as it was returning a key value pair and this [line](https://github.com/Automattic/jetpack/pull/37095/files#diff-86d81f3788f3e6c150a5c0b0671805258476cba0998eaaf019a32f4f615edfcdR100) was not working. That's why I updated it to an [object](https://github.com/Automattic/jetpack/pull/37095/files#diff-86d81f3788f3e6c150a5c0b0671805258476cba0998eaaf019a32f4f615edfcdR96) and am [looping](https://github.com/Automattic/jetpack/pull/37095/files#diff-86d81f3788f3e6c150a5c0b0671805258476cba0998eaaf019a32f4f615edfcdR104-R106) over it to match the old structure.

Reply:
Oh, I see now. `apply_filters_deprecated` doesn't actually use the new hook.